src/navigation/orbitalnavigator.cpp

```diff
@@ -897,17 +885,29 @@ void OrbitalNavigator::setAimNode(const std::string& aimNode) {
     _aim.set(aimNode);
 }
 
-void OrbitalNavigator::resetNodeMovements() {
+void OrbitalNavigator::updatePreviousAnchorState() {
     if (_anchorNode) {
         _previousAnchorNodePosition = _anchorNode->worldPosition();
         _previousAnchorNodeRotation = glm::quat_cast(_anchorNode->worldRotationMatrix());
     }
     else {
-        _previousAnchorNodePosition = glm::dvec3(0.0);
-        _previousAnchorNodeRotation = glm::dquat();
+        _previousAnchorNodePosition.reset();
```

Comment:
I like your renaming of `resetNodeMovements` to `updatePreviousAnchorState`. I prefer not using reset() here and just sticking with the original code that sets the position to zero because it's clear what is happening. A "position reset" could mean a set back to zero, or it could mean a set back to some previous value...

Reply:
I get your point, and frankly, I think the implementation of these previous positions is sort of confusing and ideally I would want to reimplement them so that it's more apparent what is going on (I started looking into that for this PR but I need to truly understand the whole update method to make sure I don't break anything, so let it go for now. But that's why there are so many changes in OrbitalNavigator in this PR)

The `_previousAnchorNodePosition` is actually a `std::optional` and is used in the updateCamera method to compute the difference between the previous position and the current one, if it has a value. So I see a danger of setting it to zero, as this could lead to big undesired differences that in turn affect the camera movement. Note that I also split up the anchor and aim reset functions and replaced code in a couple of other places in the orbitalnavigator, that used the reset version. So I wouldn't want to change it. 

However, what the `reset()` really does is just setting the value to `std::nullopt`. Maybe changing the line to 
```
_previousAnchorNodePosition = std::nullopt;
```
would be clearer?